zlib-rs/src/inflate.rs

```diff
@@ -1922,32 +1927,38 @@ pub unsafe fn inflate(stream: &mut InflateStream, flush: InflateFlush) -> Return
     if must_update_window {
         'blk: {
             // initialize the window if needed
-            if stream.state.window.size() == 0 {
-                match Window::new_in(&stream.alloc, stream.state.wbits) {
-                    Some(window) => stream.state.window = window,
+            if state.window.size() == 0 {
+                match Window::new_in(&stream.alloc, state.wbits) {
+                    Some(window) => state.window = window,
                     None => {
-                        stream.state.mode = Mode::Mem;
+                        state.mode = Mode::Mem;
                         err = ReturnCode::MemError;
                         break 'blk;
                     }
                 }
             }
 
-            stream.state.window.extend(
-                &stream.state.writer.filled()[..out_written],
-                stream.state.flags,
+            state.window.extend(
+                &state.writer.filled()[..out_written],
+                state.flags,
                 update_checksum,
-                &mut stream.state.checksum,
-                &mut stream.state.crc_fold,
+                &mut state.checksum,
+                &mut state.crc_fold,
             );
         }
     }
 
-    if let Some(msg) = stream.state.error_message {
+    if let Some(msg) = state.error_message {
         assert!(msg.ends_with(|c| c == '\0'));
         stream.msg = msg.as_ptr() as *mut u8 as *mut core::ffi::c_char;
     }
 
+    stream.data_type =
+        state.bit_reader.bits_in_buffer() as i32 +
+        if state.last { 64 } else { 0 } +
+        if matches!(state.mode, Mode::Type) { 128 } else {0} +
+        if matches!(state.mode, Mode::Len | Mode::CopyBlock) { 256 } else { 0 };
+
```

Comment:
because we're only ever in one mode, I think a `match` is nicer here. Also can tests be added that verify the behavior (probably from the libz-rs-sys` crate's test modules) versus zlib itself?

Reply:
Hmm, you're right.  I will double-check zlib's original logic and correct it to one match.  I will also put some tests together that demonstrate parity b/t libz-rs-sys and libz-sys